Automattic/vip-go-mu-plugins
# Admin Notice: add Expression_Condition, add support for WP Core classes, add a way to add notices at various stages of exectuion

## Description

This PR adds Expression_Condition (any expression that evaluates to boolean true will show the notice. 

E.g. 

```new \Automattic\VIP\Admin_Notice\Expression_Condition( $ice_cream ===  'delicious' )```

Additionally:
1. Adds support for [core classes](https://developer.wordpress.org/reference/hooks/admin_notices/) for admin notices.
2. Introduces a way to add admin notices via the new `vip_admin_notice_init` action (that fires on init). 


## Changelog Description

### Plugin Updated: Admin Notices

We've implemented changes to our Admin Notices plugin that allows for more flexibility when defining the conditions for whether a notice should be displayed or not.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Check out the PR, add the following somewhere in the app code:

```
add_action( 'vip_admin_notice_init', function( $admin_notice_controller ) {
	$admin_notice_controller->add(
		new \Automattic\VIP\Admin_Notice\Admin_Notice(
			'This ice cream has melted',
			[
				new \Automattic\VIP\Admin_Notice\Expression_Condition( 1 === 1 ),
			],
			'no-ice-cream-for-you',
			'error'
	) );
} );
```
